src/Compilers/Core/Portable/Diagnostic/TriggerDiagnosticDescriptor.cs

```diff
@@ -18,7 +18,7 @@ public class TriggerDiagnosticDescriptor : DiagnosticDescriptor
         /// <param name="customTags">Optional custom tags for the diagnostic. See <see cref="WellKnownDiagnosticTags"/> for some well known tags.</param>
         public TriggerDiagnosticDescriptor(string id, params string[] customTags)
             : base(id, title: "", messageFormat: "", category: "", defaultSeverity: DiagnosticSeverity.Hidden, isEnabledByDefault: true,
-                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable))
+                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable).Append(WellKnownDiagnosticTags.Trigger))
```

Comment:
@shyamnamboodiripad I responded.

basically this is code

internal static SimpleDiagnostic Create(string id, LocalizableString title, string category, LocalizableString message, LocalizableString description, string helpLink,
                                      DiagnosticSeverity severity, DiagnosticSeverity defaultSeverity,
                                      bool isEnabledByDefault, int warningLevel, Location location,
                                      IEnumerable<Location> additionalLocations, IEnumerable<string> customTags)
            {
                var descriptor = new DiagnosticDescriptor(id, title, message,
                     category, defaultSeverity, isEnabledByDefault, description, helpLink, customTags.ToImmutableArrayOrEmpty());
                return new SimpleDiagnostic(descriptor, severity, warningLevel, location, additionalLocations, messageArgs: null);
            }

as you can see, it always create generic DiagnosticDescriptor. now with the CustomTags.Trigger, we now have something to maintain consistency. since the "Trigger" is supposed to be used only with TriggerDiagnosticTrigger.

so, I think we should check whether given customtags have the "Trigger", and if it does, we should create TriggerDiagnosticTrigger rather than generic DiagnosticDescriptor here.